service/src/main/kotlin/fi/espoo/evaka/reports/OccupancyReport.kt

```diff
@@ -470,41 +590,41 @@ WHERE sn.placement_id = ANY(:placementIds)
         return assistanceCoefficient * serviceNeedCoefficient
     }
 
-    val placementsAndPlans = (placements + placementPlans).groupBy { it.unitId }
+    val placementsAndPlans = (placements + placementPlans).groupBy { it.groupingId }
 
-    return caretakers
-        .map { (unit, values) ->
+    return caretakerCounts
+        .map { (key, values) ->
             val occupancies = values
-                .associate { (_, date, caretakers) ->
-                    val unitPlacementsOnDate = (placementsAndPlans[unit.unitId] ?: listOf())
-                        .filter { it.period.includes(date) }
-                        .filterNot { absences[it.childId]?.any { absence -> absence.date == date } ?: false }
+                .associate { caretakers ->
+                    val unitPlacementsOnDate = (placementsAndPlans[key.groupingId] ?: listOf())
```

Comment:
is the correct name still `unitPlacementsOnDate` or can it also be `groupPlacementsOnDate` depending on type of `K`?